docs/examples/basic_meter/view.py

```diff
@@ -0,0 +1,111 @@
+# Copyright The OpenTelemetry Authors
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+"""
+This example shows how to use the different modes to capture metrics.
+It shows the usage of the direct, bound and batch calling conventions.
+"""
+from opentelemetry import metrics
+from opentelemetry.sdk.metrics import (
+    MeterProvider,
+    UpDownCounter,
+    ValueRecorder,
+)
+from opentelemetry.sdk.metrics.export import ConsoleMetricsExporter
+from opentelemetry.sdk.metrics.export.aggregate import (
+    HistogramAggregator,
+    LastValueAggregator,
+    MinMaxSumCountAggregator,
+    SumAggregator,
+)
+from opentelemetry.sdk.metrics.view import View, ViewConfig
+
+# Use the meter type provided by the SDK package
+metrics.set_meter_provider(MeterProvider())
+meter = metrics.get_meter(__name__)
+metrics.get_meter_provider().start_pipeline(meter, ConsoleMetricsExporter(), 5)
+
+requests_counter = meter.create_metric(
+    name="requests",
+    description="number of requests",
+    unit="1",
+    value_type=int,
+    metric_type=UpDownCounter,
+)
+
+requests_size = meter.create_metric(
+    name="requests_size",
+    description="size of requests",
+    unit="1",
+    value_type=int,
+    metric_type=ValueRecorder,
+)
+
+# Views are used to define an aggregation type and label keys to aggregate by
+# for a given metric
+
+# Two views with the same metric and aggregation type but different label keys
+# With ViewConfig.LABEL_KEYS, all labels but the ones defined in label_keys are
+# dropped from the aggregation
+counter_view1 = View(
+    requests_counter,
+    SumAggregator(),
+    label_keys=["environment"],
+    config=ViewConfig.LABEL_KEYS,
+)
+counter_view2 = View(
+    requests_counter,
+    MinMaxSumCountAggregator(),
+    label_keys=["os_type"],
+    config=ViewConfig.LABEL_KEYS,
+)
+# This view has ViewConfig set to UNGROUPED, meaning all recorded metrics take
+# the labels directly without and consideration for label_keys
+counter_view3 = View(
+    requests_counter,
+    LastValueAggregator(),
+    label_keys=["environment"],  # is not used due to ViewConfig.UNGROUPED
```

Comment:
If this isn't used, maybe this parameter should just be the signal that you want to use the LABEL_KEYS configuration, rather than having it only being used in that one case?

Reply:
Interesting thought. I don't know what is more confusing, the ignoring of the ViewConfig if label_keys is defined or the ignoring of the keys if ViewConfig.Ungrouped is defined.

Reply:
why not combine them into one struct that contains both the config and the data that it needs?

Reply:
Even if they are combined to one struct, i'm not sure if it eases the confusion? There's still implicit behavior that is occurring if both values are supplied, even as arguments to a struct.

Reply:
I guess I was thinking more of a class that could indicate both the type and the attributes needed to implement it, and constructors that made it clear what was required for what. Or something along those lines.